docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,317 @@
+# ImageSegmentation.jl
+
+## Introduction
+
+Image Segmentation is the process of partitioning the image into regions that have similar attributes. Image segmentation has various applications e.g, medical image segmentation, image compression and is used as a preprocessing step in higher level vision tasks like object detection and optical flow. This package is a collection of image segmentation algorithms written in Julia.
+
+## Installation
+
+```julia
+Pkg.add("ImageSegmentation")
+```
+
+## Example
+
+Let's see an example on how to use the segmentation algorithms in this package. We will try to seperate the deer, the ground and the sky in the image below using seeded region growing algorithm. This algorithm initializes the regions as the user-provided seed points and iteratively adds pixels to each region.
+
+![Original](assets/segmentation/deer.jpg) 
+
+```julia
+using Images, ImageSegmentation, ImageView
+
+img = load("deer.jpg")
```

Comment:
@annimesh2809 I would strongly encourage you to link to the Wikimedia Commons page (https://commons.wikimedia.org/wiki/File:Davidraju_Worm_Snake.jpg) instead of imgbb.com (https://ibb.co/hR2aWk) for the source of the image. That ensures the licensing information is preserved here, as required by the CC-BY-SA license (e.g as a link to the page with further information) and generally is more useful to the end user, since the image there is properly categorized, identified, etc.

In fact I'd suggest replacing all the example images by ones from Wikimedia Commons, because it's one of the few image hosting websites that make the metadata, authorship and licensing information for images very visible and convenient to access, which might come handy in cases like the one @timholy mentioned ("occasionally one gets asked to prove that you have rights to the image", etc.)

All of the images there are freely usable under open licenses, and specific images can be found easily because they tend to be well categorized. I'd be happy to help you if you need assistance locating images for any particular subject.